core/src/main/java/greencity/service/impl/SocialNetworkServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package greencity.service.impl;
+
+import greencity.entity.SocialNetwork;
+import greencity.entity.User;
+import greencity.service.SocialNetworkImageService;
+import greencity.service.SocialNetworkService;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.transaction.Transactional;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class SocialNetworkServiceImpl implements SocialNetworkService {
+    SocialNetworkImageService socialNetworkImageService;
+
+    /**
+     * Method saves all {@link SocialNetwork}.
+     *
+     * @param socialNetworkUrls socialnetwork URLs of user
+     * @param user              current user
+     * @return list of {@link SocialNetwork}
+     */
+    @Override
+    @Transactional
+    public List<SocialNetwork> saveAll(List<String> socialNetworkUrls, User user) {
+        List<SocialNetwork> listSocialNetwork = socialNetworkUrls.stream()
+            .map(url ->
+                SocialNetwork.builder()
+                    .url(url)
+                    .user(user)
+                    .socialNetworkImage(socialNetworkImageService.getSocialNetworkImageByUrl(url))
+                    .build()
+            )
+            .collect(Collectors.toList());
+        user.getSocialNetworks().clear();
```

Comment:
Please move those two lines out of this method (one level up), then:

- remove parameter user